File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.js
function solution(s){
    let arrS = s.split("")
    let ar = []
    for(a of arrS){
        if(a=="("){
            ar.push(a)
        }else if(a==")"){
            if(ar[ar.length-1]=="("){ ar.pop(a) }
            else{ ar.push(a) }
        }
    }
    if(ar.length>0){
        return false
    }
    return true
}

File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.py
def solution(s):
    arr = []
    for i in list(s):
        if(i=="("):
            arr.append(i)
        else:
            if(len(arr)>0 and arr[len(arr)-1]=="("):
                arr.pop()
            else:
                arr.append(i)
    return len(arr)==0
    